test/test_non_overlap_module.py

```diff
@@ -0,0 +1,106 @@
+'''
```

Comment:
1. Missing test case for checking if URL belongs in the format <code>/nonOverlappingModules?sem=AY%20**16/18**%20Sem%202</code>

2. Going to above URL does correctly show error message that such an AY-Semester doesn't exist, but error message is "invalid AYSem!". Please set this to something more professional, such as "The AY-Semester you specified does not exist."

Furthermore, I don't think using a script is a good way of prompting that the page with the invalid URL does not exist. Instead I would prefer that our custom 404 page is returned. However, this is merely an implementation opinion so I will not insist on having this implementation, I'll leave that decision to you entirely.

3. If I select a desired AY-Semester to view non-overlapping modules and press submit, the redirection works correctly but the <code>select</code> input field will always take on the first input which is 'AY 16/17 Sem 1'.

Manually detecting which AY-Semester was queried for and adjusting the <code>select</code> input field seems complex so I will highly recommend you NOT to do this. But at least you should leave a default placeholder input in the <code>select</code> input like 'Please choose a target AY-Semester' so that when this situation happens, it won't appear like a mismatch between what I submitted versus what I'm seeing.

4. **(Food for thought, not sure if you want to fix this)** If I want to narrow down my table view by searching for a desired module (say CS1231), the pairs where <code>string 'CS1231' < (another module name)</code> will have CS1231 all appearing on the <code>Module 2</code> column, whereas the rest where <code>string 'CS1231' >= (another module name)</code> will have CS1231 all appearing on the <code>Module 1</code> column. Is there a way to make this consistent such that for all pairs, 'CS1231' appears on one side?

5. Current description for this page is **"Shows all module pairs which no student takes together"**. I was thinking that refining it into something such as **"Shows all module pairs which no student takes together <code>in a particular semester</code>."**.

The additional text **"By default, these pairs are shown for the current AY-Semester."** can be either added as a <code>glyphicon-info-sign</code> element with <code>tooltip</code> next to the first description, or this text can be added in its entirety together with the first description into one full text.